packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -5,6 +5,8 @@
 
 import { assert } from "@fluidframework/core-utils";
```

Comment:
I'll leave generic comment:
Direction makes sense and we should do it.
But looking at "fluid:telemetry:Batching:LengthTooBig" event & Data_length > 2500, I see less than 10 events per day.
I'd think our chunking algorithm (when op grouping is not enabled) should be able to ensure that the last chunk has only op envelopes, and 2500 envelopes should fit into 1Mb, as that's ~400 bytes/op, when they should be much smaller (~250 + JSON overhead).

Are we exhausting all the opportunities here to not hit 1Mb issue? Based on Loop feedback, it feels like the answer is no. Could you please glance at that too?

Reply:
I think I see the problem with my statement. fluid:telemetry:Batching:LengthTooBig is on receiving side, not sending side